CMakeLists.txt

```diff
@@ -68,6 +70,18 @@ endif()
 
 add_feature_info(ZeroHourStuff GENZH_BUILD_ZEROHOUR "Build Zero Hour code")
 add_feature_info(GeneralsStuff GENZH_BUILD_GENERALS "Build Generals code")
+add_feature_info(InternalBuild GENZH_BUILD_INTERNAL "Building as an \"Internal\" build")
+add_feature_info(DebugBuild GENZH_BUILD_DEBUG "Building as a \"Debug\" build")
+
+add_library(gz_config INTERFACE)
+
+target_compile_options(gz_config INTERFACE ${GENZH_FLAGS})
+
+if(GENZH_BUILD_DEBUG)
+    target_compile_definitions(gz_config INTERFACE _DEBUG WWDEBUG DEBUG)
+elseif(GENZH_BUILD_INTERNAL)
+    target_compile_definitions(gz_config INTERFACE _INTERNAL)
```

Comment:
While at it, can you please also add `_RELEASE` for Release targets? Aka we want to avoid writing `defined(NDEBUG)` or `!defined(_INTERNAL) && !defined(_DEBUG)`